### PR TITLE
Feature/processing data in models

### PR DESCRIPTION
**Pull request by daudi13**
In this PR I:
- Created at least one user 
- Created at least 4 posts written by one of the users I created
- Created at least 6 posts comments for one of the posts created
- My users model includes a method that returns the 3 most recent posts for a given post
- My Post contains a method that updates the posts counter for a user and another method that returns the 5 most recent comments for a given post
- My comments model had a method that updates the comments for a post
- My Likes model has a method that updates the likes counter for a post
- Created a method called `current_user` in my ApplicationController that returns the first user for a database

---
Thank you for taking the time to review my code.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :like_count
+
+  private
+
+  def like_count
+    like.increment!(:likes_counter)
```
Comment (dasileker) [inline app/models/like.rb]: - [ ] Hello and good job on the blog app, but you have added the wrong statement for the `def like_count`, you should use post.increment instead to add like each time in the post.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  after_save :post_counter
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def post_counter
+    post.increment!(:posts_counter)
```
Comment (dasileker) [inline app/models/post.rb]: - [ ] Please refer to your diagram, the post is having an `authorid` so the right one will be adding `author.increment`.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @daudi13,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linters are ✅.
- The PR is really nice👌.

## Required Changes ♻️
_Check the comments under the review._

- [ ] Also please add relevant details to your readme file as **Getting started, the tool used, ** and add as many details to make your readme file look professional, as you can use this [ microverse readme template ](https://github.com/microverseinc/readme-template/blob/master/README.md)

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (daudi13) [reply app/models/like.rb]: thank you let me write the method.

File: app/models/post.rb (hunk 2 above)
Comment (daudi13) [reply app/models/post.rb]: Thank you @dasileker let me write the method

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi @daudi13 , :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost


 ## Status: Required Changes ♻️
- As suggested by previous reviewer, please customize your readme file to meet what you did in this project. you can follow this [template ](https://github.com/microverseinc/readme-template/blob/master/README.md)from Microverse . This make your project looks more professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by daudi13** [comment]
@uwadonat in my opinion I think this should be an optional change for now because the project is still ongoing. Unless this would have been the last segment of the project

**Review by mikethreels — APPROVED** [review]
Hi David, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**